network/src/protocols/direct_send/mod.rs

```diff
@@ -172,41 +127,20 @@ where
         );
     }
 
-    // Handle PeerNotification, which can only be NewInboundSubstream for now.
-    async fn handle_peer_notification(&mut self, notif: PeerNotification<TSubstream>) {
+    // Handle PeerNotification, which can only be NewMessage for now.
+    async fn handle_peer_notification(&mut self, notif: PeerNotification) {
         trace!("PeerNotification::{:?}", notif);
         match notif {
-            // TODO: Rate-limit number of substreams opened by a peer.
-            PeerNotification::NewSubstream(_peer_id, substream) => {
-                let ds_notifs_tx = self.ds_notifs_tx.clone();
-                self.executor.spawn(Self::handle_inbound_substream(
-                    ds_notifs_tx,
-                    self.peer_handle.peer_id(),
-                    substream.protocol,
-                    substream.substream,
-                ));
-            }
-            _ => unreachable!("Unexpected PeerNotification"),
-        }
-    }
-
-    // Handle a new inbound substream. Keep forwarding the messages to the NetworkProvider.
-    async fn handle_inbound_substream(
-        mut ds_notifs_tx: channel::Sender<DirectSendNotification>,
-        peer_id: PeerId,
-        protocol: ProtocolId,
-        substream: TSubstream,
-    ) {
-        info!(
-            "DirectSend inbound substream with peer {} for protocol {:?} opened",
-            peer_id.short_str(),
-            protocol
-        );
-        let mut substream = Framed::new(IoCompat::new(substream), LengthDelimitedCodec::new());
-        // TODO: Close inactive substreams after some duration.
-        while let Some(item) = substream.next().await {
-            match item {
-                Ok(data) => {
+            PeerNotification::NewMessage(message) => {
+                let peer_id = self.peer_handle.peer_id();
+                if let NetworkMessage::DirectSendMsg(message) = message {
```

Comment:
I'm assuming you mean `if let ` instead of `match`. I personally prefer to use match to let the compiler warn me if a variant gets added in the future that I forget to handle and to use `if let` only in very narrow cases, like handling errors/optionals which are more fixed in terms of variants. Maybe we should codify this in the Libra style guide.

Reply:
on line 136 we use a if let, I think a match would be better. It seems like we agree, so we should ideally unify on the code? if you didn't take care of this already.